### PR TITLE
Fix error in CGNS writing 

### PR DESCRIPTION
**Pull request by rois1995**
Hello everyone,

I encountered an error in the writing of a CGNS file when no nodes are associated to a given rank. It usually happens when small grids are employed and the simulation is performed on more than one core.  have uploaded at the following link the mesh and the config used to reproduce the error.

https://polimi365-my.sharepoint.com/:f:/g/personal/10507725_polimi_it/Ekl88a3uOKhMu4uCLRy3tHIBNNo6_3oMTNiZq97NB6ztDw?e=Hu8pqe

Notice that, if only one heatflux marker is selected to print the results onto, then it does not work when 2 or more cores are used. On the other hand, if both of the heatflux markers are used then it works in parallel.

To solve this issue I just added a check on the number of local points in the function CCGNSFileWriter::WriteField.

## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [X] I am submitting my contribution to the develop branch.
- [X] My contribution generates no new compiler warnings (try with --warnlevel=3 when using meson).
- [X] My contribution is commented and consistent with SU2 style (https://su2code.github.io/docs_v7/Style-Guide/).
- [ ] I used the pre-commit hook to prevent dirty commits and used `pre-commit run --all` to format old commits.
- [X] I have added a test case that demonstrates my contribution, if necessary.
- [ ] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp), if necessary.

### CODE REVIEW

File: SU2_CFD/src/output/filewriter/CCGNSFileWriter.cpp [hunk 1]
```diff
@@ -38,7 +38,7 @@ void CCGNSFileWriter::WriteData(string val_filename) {
 
   /*--- We append the pre-defined suffix (extension) to the filename (prefix) ---*/
   val_filename.append(fileExt);
-
+
```
Comment (bigfooted) [inline SU2_CFD/src/output/filewriter/CCGNSFileWriter.cpp:41]:
```suggestion

```

**Review by pcarruscag — COMMENTED** [review]
Add the case to our regressions please

**Comment by rois1995** [comment]
I think I managed to do it, please let me know if I did something wrong

File: TestCases/cgns_writer/config.cfg [hunk 2]
```diff
@@ -0,0 +1,187 @@
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%                                                                              %
+% SU2 configuration file                                                       %
+% Case description: Transonic simulation RAE2822 (RANS)                        %
```
Comment (pcarruscag) [inline TestCases/cgns_writer/config.cfg:4]: Is this right?

File: TestCases/cgns_writer/config.cfg [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+% Date: 5/15/2013                                                              %
+% File Version 7.4.0 "Blackbird"                                                %
+%                                                                              %
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+% ------------- DIRECT, ADJOINT, AND LINEARIZED PROBLEM DEFINITION ------------%
+%
+% Physical governing equations (EULER, NAVIER_STOKES,
+%                               WAVE_EQUATION, HEAT_EQUATION, FEM_ELASTICITY,
+%                               POISSON_EQUATION)
```
Comment (pcarruscag) [inline TestCases/cgns_writer/config.cfg:16]: remove comments please

File: TestCases/cgns_writer/config.cfg [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+%
+CONV_FIELD= MOMENT_X
+CONV_STARTITER= 10
+CONV_CAUCHY_ELEMS= 100
+CONV_CAUCHY_EPS= 1E-6
+
+% ------------------------- INPUT/OUTPUT INFORMATION --------------------------%
+%
+MESH_FILENAME= mesh.su2
+MESH_FORMAT= SU2
```
[the remark this answers is not in the document]
Comment (pcarruscag) [reply TestCases/cgns_writer/config.cfg:84]: Ah ok you're only testing the output, sorry

[the remark this answers is not in the document]
Comment (rois1995) [reply TestCases/cgns_writer/config.cfg:84]: Not really, it is just the output that has to be in cgns.

**Pull request merged**